src/Reader.cpp

```diff
@@ -10,13 +9,9 @@ void Reader::run() noexcept {
                 continue;
             }
             std::string str;
-            if (!std::getline(std::cin, str) || str.empty()) {
-                if (!isatty(fileno(stdin))) {
-                    return;
-                }
-                break;
+            if (std::getline(std::cin, str) && !str.empty()) {
```

Comment:
Empty lines should be accepted as possible input.

Reply:
What does empty line mean? Just '\n' symbol?

Reply:
Empty line is `""`. It will be returned by `getline()` in case of no characters between previous and next ends of lines.

Reply:
Ok, but why we need trigger on it?
Empty line is not interrupt the program execution yet.

Reply:
Empty string should be appened to output with space. So, for this input:

```
1

2
```

should be generated this output (two spaces appears here):

```
1  3
```

Reply:
Oh, ok. I got your example.

Reply:
I thought that was the wrong behavior. :)